### PR TITLE
boot: reseal the run key for all recovery systems, but recovery keys only for the good ones

### PR DESCRIPTION
**Pull request by bboozzoo**
Tweak the inputs for resealing the keys. With introduction of a tried recovery
system tracked in the current recovery systems list, it was decided that only
the run key shall be resealed for those systems. The recovery keys, are resealed
for a list of known good recovery systems.

### CODE REVIEW

File: boot/seal.go [hunk 1]
```diff
@@ -368,7 +368,23 @@ func resealKeyToModeenvSecboot(rootdir string, model *asserts.Model, modeenv *Mo
...
+	if err != nil {
+		return fmt.Errorf("cannot compose recovery boot chains for run key: %v", err)
+	}
+
+	// the boot chains for recovery keys include only those system that were
+	// tested and are known to be good
+	testedRecoverySystems := modeenv.GoodRecoverySystems
+	if len(testedRecoverySystems) == 0 && len(modeenv.CurrentRecoverySystems) > 0 {
+		// compatibility for systems where good recovery systems list
+		// has not been populated yet
```
Comment (pedronis) [inline boot/seal.go:384]: we might want to log something here

**Review by pedronis — APPROVED** [review]
lgtm

**Review by mvo5 — APPROVED** [review]
Thank you!

**Review by anonymouse64 — APPROVED** [review]
LGTM, thanks for this

**Comment by bboozzoo** [comment]
The spread run finished with bunch of unrelated failures in ests/main/apt-hooks and tests/main/snap-advise-command.

**Pull request merged**